### PR TITLE
[NETBEANS-2925] Possible solution to not to filter out sources from the buildDir

### PR DESCRIPTION
**Pull request by lkishalmi**
[NETBEANS-2925](https://issues.apache.org/jira/browse/NETBEANS-2925)

Well, I need some advice here. Returning mixed on the contents of build dir is probably not the best thing to do. Shall I initiate a different package view with different filter for generated sources instead?

### CODE REVIEW

**Comment by jglick** [comment]
I am not sure what you are trying to do but I am pretty sure this is not the right way to do it.

The `maven` module displays package roots for any `build/generated-{,test-}sources/*/` it finds. The fact that these are `NOT_SHARABLE` is irrelevant.

**Comment by lkishalmi** [comment]
Thanks, Jesse!
Checked the Maven implementation and made something similar in Gradle as well.

File: groovy/gradle.java/src/org/netbeans/modules/gradle/java/classpath/GradleSourcesImpl.java [hunk 1]
```diff
@@ -162,13 +164,7 @@ public GradleSourcesImpl(Project proj) {
...
                 }
             }
-            Collections.sort(ret, new Comparator<SourceGroup>() {
-
-                @Override
-                public int compare(SourceGroup o1, SourceGroup o2) {
-                    return o1.getName().compareTo(o2.getName());
-                }
-            });
+            Collections.sort(ret, (SourceGroup o1, SourceGroup o2) -> o1.getName().compareTo(o2.getName()));
```
Comment (jglick) [inline groovy/gradle.java/src/org/netbeans/modules/gradle/java/classpath/GradleSourcesImpl.java]:
```suggestion
            Collections.sort(ret, Comparator.comparing(SourceGroup::getName));
```

or something like that (look it up).

Comment (lkishalmi) [reply groovy/gradle.java/src/org/netbeans/modules/gradle/java/classpath/GradleSourcesImpl.java]: Thanks! Learn a new thing every day!
The test will be a new PR as it is not that trivial to create one for this.

**Pull request merged**